_posts/2017-01-30-year-in-art.markdown

```diff
@@ -0,0 +1,66 @@
+---
+layout: post
+title: "Animating the Year In Art"
+date: 2017-01-30 14:21
+author: eve
+categories: [javascript]
+---
+
+Artsy's end of year features are and annual chance to walk through highlights of the year while also exploring front-end experiments. Created in collaboration with UBS and designed by Owen Dodd, [The Year In Art 2016](http://artsy.net/year-in-art-2016) presents an interactive timeline of singular moments in art and culture over the past year.
+
+The piece opens with header animation, a series of transparent sliding boxes that presented a unique challenge. The finalized look is somewhat like a slinky-— a stack of containers that are stretched open from the bottom, and compress again as they reach the top of the viewport, collapsing inward without ever crossing outside the screen.
+
+Achieving this effect required animating elements in response both to the size of other elements in the viewport, and to the client’s scroll interactions.
+
+<!-- more -->
+
+Because each element in the animation opens out from the center, and because the entire container is transparent, the sections needed to adjust their height rather than simply sliding out of view. Furthermore, the entire animation needed be responsive, and able to it’s place easily on page when it's refreshing or resized.
+
+Essentially we needed a way to know where in our scroll position the top of each section 'enters' the window, which is also the point at which it should become visible. This functionality is close to Waypoints, a library we use often at Artsy, but that option wasn't going to work in this case. Because each section was actually in the viewport already, just at 0px tall, the library wouldn’t recognize an element's ‘entering’ the frame (because it’s already there).
```

Comment:
Could be worth linking to the Waypoints GitHub page and maybe even a part of force that uses it.